packages/aws_common/lib/src/logging/aws_logger.dart

```diff
@@ -63,6 +69,20 @@ class AWSLogger implements Closeable {
 
   final Logger _logger;
 
+  /// Parent of this logger in the logger hierarchy.
+  AWSLogger? get _parent {
+    return activeLoggers[_logger.parent?.fullName];
+  }
+
+  /// Children of this logger in the logger hierarchy.
+  List<AWSLogger> get _children {
```

Comment:
This can probably be optimized -- if every time you create logger, you add the instance to its parent's `children` array, this could be static instead of built each time it's needed